demos/python_demos/asl_recognition_demo/asl_recognition_demo/visualizer.py

```diff
@@ -0,0 +1,112 @@
+"""
+ Copyright (c) 2019 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import time
+from multiprocessing import Process, Queue, Value
+
+import cv2
+import numpy as np
+
+class Visualizer:
+    """Class that allows to play video sources with different speed"""
+
+    def __init__(self, trg_fps=60):
+        """Constructor"""
+
+        self._trg_time_step = 1. / float(trg_fps)
+        self._last_key = Value('i', -1, lock=True)
+        self._need_stop = Value('i', False, lock=False)
+        self._worker_process = None
+        self._tasks = dict()
+
+    def register_window(self, name):
+        """Allocates resources for the new window"""
+
+        if self._worker_process is not None and self._worker_process.is_alive():
+            raise EnvironmentError('Cannot add the window for running visualizer')
+
+        self._tasks[name] = Queue(1)
+
+    def get_key(self):
+        """Returns the value of pressed key"""
+
+        out_key = self._last_key.value
+        if out_key != -1:
+            with self._last_key.get_lock():
```

Comment:
This `with` should contain the `out_key` assignment as well, since otherwise it's possible to miss keypresses that happen between the read and write of `self._last_key.value`.

Reply:
Moved.